### ISSUE TITLE
Why does model.predict() have self.eval() and self.train() ?

### ISSUE THREAD

**Issue by triple-tam** [chenyuntc/simple-faster-rcnn-pytorch#154]
In fasterRCNN.py, the model has a predict() function - which I understand to be the model making predictions on input images based on its current trained state. If it should simply do a forward pass, why is self.eval() and self.train() in the predict() function? Or is my understanding incorrect?

Thank you,
Tam

**Comment by triple-tam** [comment]
So if there are any other deep learning newbies like me, this might help:
- Self eval and self train turn off/on dropouts and batchnorm stat freezing
- eval() is turned on so that batchnorm stats are frozen and dropout isn't used anymore, then does "predict" stuff, and then turns on train() to have everything how it was before
- sometimes other codes write model.train() and model.eval() in the train.py file; it is a matter of style
